src/jenkins.ts

```diff
@@ -0,0 +1,243 @@
+   *
+   * @default - { name: 'kubernetes', version: '1.31.3' }, { name: 'workflow-job', version: '1145.v7f2433caa07f' }, { name: 'workflow-aggregator', version: '2.6' }, { name: 'git', version: '4.10.3' }, { name: 'job-dsl', version: '1.78.1' }, { name: 'configuration-as-code', version: '1414.v878271fc496f' }, { name: 'kubernetes-credentials-provider', version: '0.20' }
+   */
+  readonly basePlugins?: Plugin[];
+  /**
+   * List of custom plugins applied to Jenkins resource.
+   *
+   * @default - []
+   */
+  readonly plugins?: Plugin[];
+  /**
+   * List of seed job configuration for Jenkins resource.
+   */
+  readonly seedJobs?: SeedJob[];
+}
+
+/**
+ * A jenkins instance.
+ */
+export class Jenkins extends Construct {
+  private _plugins: Plugin[];
+  private _basePlugins: Plugin[];
+  private _seedJobs: SeedJob[];
+
+  constructor(scope: Construct, id: string, props: JenkinsProps = {}) {
+    super(scope, id);
+
+    const namespace = props.namespace ?? 'default';
+    const labels = props.labels ?? { app: 'jenkins' };
+    const authorizationStrategy = 'createUser';
+    const disableCSRFProtection = props.disableCsrfProtection ?? false;
+    this._basePlugins = getJenkinsBasePlugins(props.basePlugins);
+    this._plugins = props.plugins ?? [];
+    const containers = getContainerConfiguration();
+    this._seedJobs = props.seedJobs ?? [];
+
+    const jenkinsCustomResource = new JenkinsCustomResource(this, 'JenkinsCustomResource', {
+      metadata: {
+        namespace: namespace,
+        labels: labels,
+        name: 'cdk8s-jenkins-construct',
+      },
+      spec: {
+        groovyScripts: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        configurationAsCode: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        jenkinsApiSettings: {
+          authorizationStrategy: authorizationStrategy,
+        },
+        master: {
+          disableCsrfProtection: disableCSRFProtection,
+          containers: containers,
+          basePlugins: this._basePlugins,
+          plugins: this._plugins,
+        },
+        seedJobs: this._seedJobs,
+      },
+    });
+
+    this.node.defaultChild = jenkinsCustomResource;
+  }
+
+  /**
+   * Add base plugins to jenkins instance.
+   * @param basePlugins List of base plugins.
+   */
+  public addBasePlugins(...basePlugins: Plugin[]): void {
+    this._basePlugins.push(...basePlugins);
+  }
+
+  /**
+   * Add custom plugins to jenkins instance.
+   * @param plugins List of custom plugins.
+   */
+  public addPlugins(...plugins: Plugin[]): void {
+    this._plugins.push(...plugins);
+  }
+
+  /**
+   * Add seed jobs to jenkins instance.
+   * @param seedJobs List of seed jobs.
+   */
+  public addSeedJobs(...seedJobs: SeedJob[]): void {
+    this._seedJobs.push(...seedJobs);
+  }
+}
+
+function getJenkinsBasePlugins(basePluginsUpdates: Plugin[] | undefined): Plugin[] {
```

Comment:
```suggestion
function getJenkinsBasePlugins(basePluginsUpdates?: Plugin[]): Plugin[] {
```

Reply:
Removing this function in favor of using `addBasePlugins` in the commit.